packages/css/checkbox.css

```diff
@@ -125,28 +125,28 @@
   /* Only use hover for non-touch devices to prevent sticky-hovering
     "input:not(:read-only)" does not work so using ".container:not(.readonly) >" instead */
   @media (hover: hover) and (pointer: fine) {
-    .fds-checkbox:not(.fds-checkbox--readonly, .fds-checkbox--disabled) > .fds-checkbox__label:hover,
-    .fds-checkbox:not(.fds-checkbox--readonly, .fds-checkbox--disabled) > .fds-checkbox__input:hover + .fds-checkbox__label {
+    .fds-checkbox:not(.fds-checkbox--readonly) .fds-checkbox__input:not(:disabled) ~ .fds-checkbox__label:hover,
+    .fds-checkbox:not(.fds-checkbox--readonly) .fds-checkbox__input:hover:not(:disabled) ~ .fds-checkbox__label {
       color: var(--fds-semantic-text-action-hover);
     }
 
-    .fds-checkbox:not(.fds-checkbox--readonly, .fds-checkbox--disabled) > .fds-checkbox__input:hover:not(:checked) {
+    .fds-checkbox:not(.fds-checkbox--readonly) .fds-checkbox__input:hover:not(:checked, :disabled) {
       --fds-checkbox-border-color: var(--fds-semantic-border-input-hover);
 
       box-shadow:
         var(--fds-checkbox-border__hover),
         inset 0 0 0 2px var(--fds-checkbox-border-color);
     }
```

Comment:
Could we solve `readonly` by checking `.fds-checkbox:not([readonly])`, or add `aria-readonly="true/false"`?
I think it would be nice if we can eliminate this class, and only use data-attributes or selectors

Reply:
> Could we solve readonly by checking .fds-checkbox:not([readonly]),

I was thinking the samething! 💡 This would make it so much easier to work with, but Readonly is only supported by input types that can change/mutate its value. Checkboxes apparently have two fixed values check or not. 

> or add aria-readonly="true/false"?

It is used in some places already (also in css). I was thinking about using the aria attr too glad you brought it up. 
The downside of this is that the classes might be a little more complex to understand, and its a bit more work to add everywhere.

Nice suggestion, think I'll start with adding aria-readonly attributes and connect the css to those attributes.

Reply:
Lets try these out :) 

I suggest however we also do some research on whats the best practice here, and attr selectors also have worse performance than pure classes. I am a bit worried about the longterm development and maintenance for using attr selectors as we are breaking seperation of concern with using attr selectors as we making a hard-link between styling and html. There might be edge-cases where this is might be a problem and also limits composability of the css.

Reply:
Readonly-state is a tough one because as mentioned sometimes is enable by `readonly`, other times `aria-readonly` depending on the html element. 🤔 

Did a quick search on support for using `aria-readonly` and SR, might be a bit outdated by its things to test for: https://adrianroselli.com/2022/11/brief-note-on-aria-readonly-support-html.html

Reply:
We could always do two selectors for a style 🤔 
```css
.fds-checkbox:not([readonly]),
.fds-checkbox:not([aria-readonly]) {
...
}

Reply:
Suggest just keeping it as a `readonly` class for now, we can always change it later if we find out thats a better solution. I am however not sure its the best approach so lets save it for #2065.